src/compiler/checker.ts

```diff
@@ -923,7 +924,9 @@ namespace ts {
                     if (!target.exports) target.exports = createSymbolTable();
                     mergeSymbolTable(target.exports, source.exports);
                 }
-                recordMergedSymbol(target, source);
+                if (!unidirectional) {
+                    recordMergedSymbol(target, source);

```

Comment:
@sandersn you were right that the merged symbols essentially go both ways (target and source don't share a mergeId, but target gets there by `cloneSymbol(resolvedTarget)` and being mutated; then source gets added here.) Introducing a `unidirectonal` option to prevent adding a `mergeId` to `source` accomplishes your suggestion.

Reply:
Hm. I've been burned pretty badly in the past by trying to extend the architecture of symbol merging. Merging is where state touches the checker, which is otherwise stateless: 
Symbols from the binder are created when loading a file and only re-created when that file is editted. But merging mutates those symbols. It's important to only do that once, and tricky to make sure that happens as checkers are re-created over and over again and run merge* each time. I'd rather not change the architecture if it's not necessary.

Can you instead investigate copying constituent symbols from `mainModule.exports` and `moduleAugmentation.exports` to a fresh symbol created from `cloneSymbol(moduleAugmentation)` (or maybe it's called createSymbol)? This method is more likely to break symbol sharing, so you'll need to add tests for find-all-refs and rename-symbol to make they work correctly.

The patternAmbientModuleAugmentation map is still probably needed.

Reply:
Not sure I follow—`moduleAgumentation` (the target) isn't transient, so `mergeSymbol` clones it rather than mutates it, and `mainModule` (the source) doesn't get mutated because it's the source. I _think_ the only thing getting mutated here is the `mergedSymbols` array, which is the problem—I need to avoid calling `recordMergedSymbol` with these two symbols, because you should never be able to get from `mainModule` to `moduleAugmentation` via `getMergedSymbol`.